### PR TITLE
luhn: update to latest canonical tests

### PR DESCRIPTION
**Pull request by stkent**
https://github.com/exercism/x-common/blob/134e4970dfa3f1cee7e054ddcacf694609f19290/exercises/luhn/canonical-data.json

Also includes some adjustments to the example solution to fix broken edge cases.

### CODE REVIEW

File: exercises/luhn/src/test/java/LuhnValidatorTest.java [hunk 1]
```diff
@@ -19,12 +19,24 @@ public void testThatTheStringConsistingOfASingleZeroIsInvalid() {
 
     @Ignore
     @Test
-    public void testThatAValidCanadianSocialInsuranceNumberIsIdentifiedAsValid() {
+    public void testThatASimpleValidCanadianSocialInsuranceNumberIsIdentifiedAsValid() {
```
Comment (FridaTveit) [inline exercises/luhn/src/test/java/LuhnValidatorTest.java]: Not that it matters very much, but I don't think this is a valid canadian SIN as it only has two digits. In the canonical data this test is called "simple valid sin" so maybe change this test name to `testThatASimpleValidSocialInsuranceNumberIsIdentifiedAsValid()`?

Comment (stkent) [inline exercises/luhn/src/test/java/LuhnValidatorTest.java]: Good catch; will update, thanks!

**Comment by stkent** [comment]
@FridaTveit updated!

**Comment by FridaTveit** [comment]
Great, thanks! :)
